tests/testtcptransport.nim

```diff
@@ -174,6 +174,23 @@ suite "TCP transport":
 
   proc transProvider(): Transport = TcpTransport.new(upgrade = Upgrade())
 
+  asyncTest "Custom timeout":
```

Comment:
Could you please explain how this test is checking if the timeout is working properly and in which scenario it could fail?

Reply:
The test is waiting for the connection to be closed by the timeout)